go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/sigstore/sigstore
 
-go 1.22.8
+go 1.22.0
```

Comment:
we will need to update places like https://github.com/sigstore/sigstore/blob/main/.github/workflows/e2e_test.yml#L37C1-L37C34 to pin to `1.22` or `1.23` otherwise it will install the 1.22.0 only and i think we want to use the latest available always

Reply:
cc @haydentherapper

Reply:
Good spot, pushed up [d8cb131](https://github.com/sigstore/sigstore/pull/1878/commits/d8cb131efa0f6cb70b4f8f192adcd89c2279b216) as a suggestion of how to address that

Reply:
Caveat: I'm not an expert on Go version resolution

The reason we began to include the patch release version is because the tooling required us to either set the specific version of Go or add a `toolchain` directive. When @cpanato started moving us to 1.23, it seemed like something changed in the tooling and we didn't need to specify the patch version, only the minor version like what we used to do.

If it is possible to only specify `1.22` or `1.23`, let's do that. I'd rather not have the complexity of CI tooling pulling a different version than what's in the go.mod file.

Reply:
Certainly you can't use `go 1.22` in go.mod, in recent go1.22.x versions they'll just complain that go.mod needs tidying and will update it to `1.22.0` and also add a `toolchain go1.22.8` directive:
```
% grep -E '^(go|toolchain)' go.mod
go 1.22
% go1.22.8 build ./...
go: updates to go.mod needed; to update it:
        go mod tidy
% go1.22.8 mod tidy
% grep -E '^(go|toolchain)' go.mod
go 1.22.0
toolchain go1.22.8
```

Reply:
Something does seem to have changed in 1.23 whereby '1.23' is seemingly accepted as a directive again

```
% grep -E '^(go|toolchain)' go.mod
go 1.23
% go1.23.2 build ./...
% go1.23.2 mod tidy
% grep -E '^(go|toolchain)' go.mod
go 1.23
```

Reply:
The change in behaviour in 1.23 is referenced in this comment on this well-cited GH issue on the confusion around the go directive changes:

![image](https://github.com/user-attachments/assets/cd30e750-f034-4d88-896d-5d408d900b86)

https://github.com/golang/go/issues/62278#issuecomment-2062002018

However, that's really saying that `go` itself will just silently assume 1.23.0 for module requirements and toolchain rather than the more useful 1.23.X. 

Unfortunately from a GitHub Actions perspective, setup-go doesn't currently know or use go toolchain versions, it's just manually parsing the go.mod for 'go' directives itself and hasn't really kept up with the 1.22 and 1.23 changes here. There's an issue tracking that over [here](https://github.com/actions/setup-go/issues/457)

Reply:
thank you @dnwe for all the clarification TIL a lot about this fuzzy thing that i cant understand

Reply:
Thanks for digging into this!